### DISCUSSION TITLE
Iterating over groups - without DataFrame

### DISCUSSION THREAD

**Discussion by david32** [danielhrisca/asammdf#643] [Ideas]
What is the reason that its only possible to iterate over groups by getting a pandas DataFrame?

Basically why isn't it possible to iterate over groups receiving a "Channel Group" object  (I know it currently doesn't formally exist) and working with Groups? 
It is possible to display a group, or export a group in the GUI, but no way to iterate and manipulate it using the library?
[for example removing a group, or replacing a group with a modified version of it]

**Comment by danielhrisca** [comment]
What exactly do you want to achieve with the ``Channel group`` object? What would it contain?

**Reply by david32** [reply]
Well, 
Just extra control I guess, especially on deletion.
If we look, there is a sort of group creation when appending a list of signals. But nothing for editing or deleting groups.
The class already exists.
